File: frontend/src/utils/weight.ts
/** Parst eine Gewichts-Nutzereingabe (Komma ODER Punkt als
 * Dezimaltrennzeichen, z.B. "76,05" oder "76.05") zu einer Zahl,
 * gerundet auf die naechsten 0,05 kg - siehe Design-Spec
 * "Usability-Fixes Runde 2" Abschnitt 2. Gibt null bei leerem Input
 * zurueck (== "kein Wert eingegeben"), NaN bei tatsaechlich ungueltigem
 * Input (Aufrufer soll das von "kein Wert" unterscheiden koennen). */
export function parseWeightInput(raw: string): number | null {
  const trimmed = raw.trim();
  if (trimmed === "") return null;
  const normalized = trimmed.replace(",", ".");
  const value = Number(normalized);
  if (!Number.isFinite(value)) return NaN;
  return Math.round(value / 0.05) * 0.05;
}

File: backend/app/utils/weight.py
"""Gespiegelte Logik zu frontend/src/utils/weight.ts - Komma/Punkt als
Dezimaltrennzeichen tolerieren, auf 0,05kg-Schritte runden. Siehe
Design-Spec "Usability-Fixes Runde 2" Abschnitt 2. Als Defensive-
Maßnahme gedacht (v.a. für Endpunkte, die direkt per API ohne unser
Frontend aufgerufen werden koennten) - das Frontend normalisiert schon
selbst, bevor es sendet."""


def parse_weight_kg(value: float | str | None) -> float | None:
    """Akzeptiert float, int, oder String mit Komma/Punkt. Gibt None bei
    None/leerem String zurueck. Wirft ValueError bei nicht parsbarem
    String (Aufrufer entscheidet, wie das dem Nutzer gemeldet wird)."""
    if value is None:
        return None
    if isinstance(value, str):
        trimmed = value.strip()
        if trimmed == "":
            return None
        value = float(trimmed.replace(",", "."))
    return round(float(value) / 0.05) * 0.05
